### ISSUE TITLE
new expression being used in src/zephyr/concrete_ble_transmitter.cpp

### ISSUE THREAD

**Issue by liammcnair-fdh** [vmware/herald-for-cpp#52]
I am seeing the `new` expression being used in the following line:
```
src/ble/zephyr/concrete_ble_transmitter.cpp:144:      char* newvalue = new char[payload->size()];
```
I don't believe zephyr supports using the `new` operator as per documentation [here](https://docs.zephyrproject.org/latest/reference/kernel/other/cxx_support.html#c-support-for-applications).

**Comment by adamfowleruk** [comment]
It does, but are right in raising it. We're trying to remove all dynamic allocation (which will also mean removing most STL use).

**Comment by adamfowleruk** [comment]
Note: Zephyr docs do not reflect current feature status. See https://github.com/zephyrproject-rtos/zephyr/issues/31281

**Comment by adamfowleruk** [comment]
This is correct usage according to the samples, but it was not clearing up the newvalue correctly and so leaking memory over time. A fix for this will be committed momentarily. newvalue dynamic allocation is required as the payload changes over time, and the length could change too. Once read into the attribute cache in Zephyr the value can safely be deleted.
